kafka/producer.go

```diff
@@ -236,10 +249,8 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 			ctx = queuecontext.DetachedContext(ctx)
 		}
 		p.client.Produce(ctx, record, func(msg *kgo.Record, err error) {
-			defer func() {
-				wg.Done()
-				span.End()
-			}()
+			defer wg.Done()
+
 			if err != nil {
 				span.RecordError(err)
 				span.SetStatus(codes.Error, err.Error())
```

Comment:
**question**: if the span ended and production is performed async, can we still change the status and record the error here ? I don't think that's possible and we might need a child span

Reply:
This could be fixed with a `Produce` span, which tracks our own method (not the kgo internal ones, so it's different).
Then, the span isn't finished when this happens async.

Reply:
We'd probably want to call `span.End()` for each of the `Produce` spans on the `Produce callback` that `kgo` calls after the record is produced.

Reply:
Isn't that what we already do?
Now that you mention it though, I wonder if we should even track errors in this callback. kgo will already record the error in its own hook.

So couldn't we entirely ignore that error?